### PR TITLE
update compileKokkosKernels* scripts

### PR DESCRIPTION
**Pull request by ndellingwood**

### CODE REVIEW

**Comment by ndellingwood** [comment]
These are example scripts for compiling/install kokkos-kernels, they are not tested by cm_test_all_sandia so I won't run spot-checks on the changes. I confirmed the cmake changes to compileKokkosKernels.sh worked locally (with necessary changes to paths and my laptops hardware etc).

File: example/buildlib/compileKokkosKernels.sh [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
+# Configure Kokkos (Unit Tests OFF) - Makefile located in kokkos-build
+cmake -Bkokkos-build -DCMAKE_CXX_COMPILER=${CXX} -DKokkos_ARCH_PASCAL60=ON -DKokkos_ARCH_POWER8=ON -DKokkos_ENABLE_CUDA=ON -DKokkos_ENABLE_SERIAL=ON -DKokkos_ENABLE_OPENMP=ON -DKokkos_ENABLE_CUDA_LAMBDA=ON -DCMAKE_CXX_FLAGS="${CXXFLAGS}" -DCMAKE_INSTALL_PREFIX="${PWD}/kokkos-install" -DKokkos_ENABLE_TESTS=OFF ${KOKKOS_PATH}
+
+# Build and Install Kokkos - install lib at ${PWD}/kokkos-install
+cmake --build kokkos-build -j 8 --target install
+
+
+# Configure KokkosKernels (Unit Tests OFF) - Makefile located in kokkoskernels-build
+cmake -Bkokkoskernels-build -DCMAKE_CXX_COMPILER=${CXX} -DKokkos_ROOT="${PWD}/kokkos-install" -DKokkosKernels_INST_DOUBLE=ON -DKokkosKernels_INST_COMPLEX_DOUBLE=ON -DKokkosKernels_INST_ORDINAL_INT=ON -DKokkosKernels_INST_ORDINAL_INT64_T=ON -DKokkosKernels_INST_OFFSET_INT=ON -DKokkosKernels_INST_OFFSET_SIZE_T=ON -DKokkosKernels_INST_LAYOUTLEFT=ON -DKokkosKernels_ADD_DEFAULT_ETI=ON -DCMAKE_INSTALL_PREFIX="${PWD}/kokkoskernels-install" -DKokkosKernels_ENABLE_TESTS=OFF  ${KOKKOSKERNELS_PATH}
```
Comment (e10harvey) [inline example/buildlib/compileKokkosKernels.sh:19]:
```suggestion
cmake -Bkokkoskernels-build -DCMAKE_CXX_COMPILER=${CXX} -DKokkos_ROOT="${PWD}/kokkos-install" -DKokkosKernels_INST_DOUBLE=ON -DKokkosKernels_INST_COMPLEX_DOUBLE=ON -DKokkosKernels_INST_ORDINAL_INT=ON -DKokkosKernels_INST_ORDINAL_INT64_T=ON -DKokkosKernels_INST_OFFSET_INT=ON -DKokkosKernels_INST_OFFSET_SIZE_T=ON -DKokkosKernels_INST_LAYOUTLEFT=ON -DKokkosKernels_ADD_DEFAULT_ETI=ON -DCMAKE_INSTALL_PREFIX="${PWD}/kokkoskernels-install" -DKokkosKernels_ENABLE_TESTS=OFF -DKokkosKernels_ENABLE_TPL_CUBLAS=OFF ${KOKKOSKERNELS_PATH}
```

File: example/buildlib/compileKokkosKernels.sh [hunk 2]
```diff
@@ -1,13 +1,22 @@
-KOKKOS_PATH=${HOME}/work/kokkos #path to kokkos source
-KOKKOSKERNELS_SCALARS='double,"complex<double>"' #the scalar types to instantiate =double,float...
-KOKKOSKERNELS_LAYOUTS=LayoutLeft #the layout types to instantiate.
-KOKKOSKERNELS_ORDINALS=int,long #ordinal types to instantiate
-KOKKOSKERNELS_OFFSETS=int,size_t #offset types to instantiate
-KOKKOSKERNELS_PATH=../.. #path to kokkos-kernels top directory.
-KOKKOSKERNELS_OPTIONS=eti-only #options for kokkoskernels  
+#!/bin/bash
+
```
Comment (e10harvey) [inline example/buildlib/compileKokkosKernels.sh:2]:
```suggestion
# Requires cmake version > 3.12
```

File: example/buildlib/compileKokkosKernelsSimple.sh [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
+CXX=${KOKKOS_PATH}/bin/nvcc_wrapper
 KOKKOSKERNELS_OPTIONS=eti-only #options for kokkoskernels  
-KOKKOS_DEVICES=Cuda # other devices Cuda,Serial ..
+KOKKOS_DEVICES=Cuda
 KOKKOS_ARCHS=SKX,Volta70
+KOKKOS_CUDA_OPTIONS=enable_lambda
 CXXFLAGS="-Wall -pedantic -Werror -O3 -g -Wshadow -Wsign-compare -Wtype-limits -Wuninitialized"
 
-../../scripts/generate_makefile.bash --kokkoskernels-path=${KOKKOSKERNELS_PATH} --with-scalars=${KOKKOSKERNELS_SCALARS} --with-ordinals=${KOKKOSKERNELS_ORDINALS} --with-offsets=${KOKKOSKERNELS_OFFSETS} --kokkos-path=${KOKKOS_PATH} --with-devices=${KOKKOS_DEVICES} --arch=${KOKKOS_ARCHS} --compiler=${CXX} --with-options=${KOKKOSKERNELS_OPTIONS}  --cxxflags="${CXXFLAGS}"
+../../scripts/cm_generate_makefile.bash --kokkoskernels-path=${KOKKOSKERNELS_PATH} --with-scalars=${KOKKOSKERNELS_SCALARS} --with-ordinals=${KOKKOSKERNELS_ORDINALS} --with-offsets=${KOKKOSKERNELS_OFFSETS} --kokkos-path=${KOKKOS_PATH} --with-devices=${KOKKOS_DEVICES} --arch=${KOKKOS_ARCHS} --compiler=${CXX} --with-cuda-options=${KOKKOS_CUDA_OPTIONS} --with-options=${KOKKOSKERNELS_OPTIONS} --cxxflags="${CXXFLAGS}"
```
Comment (e10harvey) [inline example/buildlib/compileKokkosKernelsSimple.sh:15]:
```suggestion
../../cm_generate_makefile.bash --kokkoskernels-path=${KOKKOSKERNELS_PATH} --with-scalars=${KOKKOSKERNELS_SCALARS} --with-ordinals=${KOKKOSKERNELS_ORDINALS} --with-offsets=${KOKKOSKERNELS_OFFSETS} --kokkos-path=${KOKKOS_PATH} --with-devices=${KOKKOS_DEVICES} --arch=${KOKKOS_ARCHS} --compiler=${CXX} --with-cuda-options=${KOKKOS_CUDA_OPTIONS} --with-options=${KOKKOSKERNELS_OPTIONS} --cxxflags="${CXXFLAGS}"
```

**Review by e10harvey — CHANGES_REQUESTED** [review]
Thanks, @ndellingwood! I found a few problems when testing the changes.

File: example/buildlib/compileKokkosKernels.sh (hunk 1 above)
Comment (ndellingwood) [reply example/buildlib/compileKokkosKernels.sh:19]: @e10harvey what version of Cuda were you testing with that required disabling CUBLAS, and should I also update the cmake line to disable CUSPARSE as well?

Comment (e10harvey) [reply example/buildlib/compileKokkosKernels.sh:19]: 10.1. I think just CUBLAS needs to be disabled.

**Comment by ndellingwood** [comment]
@e10harvey thanks for review! I committed your suggestions and rebased down to a single commit, this should be ready for merge I think

**Review by e10harvey — APPROVED** [review]

**Pull request merged**